File: classes/en/Comercio.php
<?php 
   // require_once '../../recursos/compartidos/inclusiones.php';
    require_once $RM_ENTITIES_DIR.'Galeria.php';
    require_once $RM_ENTITIES_DIR.'Geolocalizacion.php';
    require_once $RM_ENTITIES_DIR.'Preferencias.php';
    require_once $RM_ENTITIES_DIR.'CategoriaComercio.php';
    require_once $RM_ENTITIES_DIR.'Usuario.php';
    require_once $RM_ENTITIES_DIR.'Subscripcion.php';
    require_once $RM_ENTITIES_DIR.'Producto.php';
    require_once $RM_ENTITIES_DIR.'Promocion.php';
    require_once $RM_ENTITIES_DIR.'SolicitudCliente.php';


class EN_Comercio{
var $idComercio;
var $noComercio;
var $diComercio;
var $prComercio;
var $t1Comercio;
var $t2Comercio;
var $emComercio;
var $imagenes;
var $geolocalizaciones;
var $preferencias;
var $clasificaciones;
var $subcripciones;
var $promociones;
var $productos;
var $solicitudes;
    
function setImagenes($imagenes) { $this->imagenes = $imagenes; }
function getImagenes() { return $this->imagenes; }
function setGeolocalizaciones($geolocalizaciones) { $this->geolocalizaciones = $geolocalizaciones; }
function getGeolocalizaciones() { return $this->geolocalizaciones; }
function setPreferencias($preferencias) { $this->preferencias = $preferencias; }
function getPreferencias() { return $this->preferencias; }
function setClasificaciones($clasificaciones) { $this->clasificaciones = $clasificaciones; }
function getClasificaciones() { return $this->clasificaciones; }
function setSubcripciones($subcripciones) { $this->subcripciones = $subcripciones; }
function getSubcripciones() { return $this->subcripciones; }
function setIdComercio($idComercio) { $this->idComercio = $idComercio; }
function getIdComercio() { return $this->idComercio; }
function setNoComercio($noComercio) { $this->noComercio = $noComercio; }
function getNoComercio() { return $this->noComercio; }
function setDiComercio($diComercio) { $this->diComercio = $diComercio; }
function getDiComercio() { return $this->diComercio; }
function setPrComercio($prComercio) { $this->prComercio = $prComercio; }
function getPrComercio() { return $this->prComercio; }
function setT1Comercio($t1Comercio) { $this->t1Comercio = $t1Comercio; }
function getT1Comercio() { return $this->t1Comercio; }
function setT2Comercio($t2Comercio) { $this->t2Comercio = $t2Comercio; }
function getT2Comercio() { return $this->t2Comercio; }
function setEmComercio($emComercio) { $this->emComercio = $emComercio; }
function getEmComercio() { return $this->emComercio; }
function setPromociones($promociones) { $this->promociones = $promociones; }
function getPromociones() { return $this->promociones; }
function setProductos($productos) { $this->productos = $productos; }
function getProductos() { return $this->productos; }
function setSolicitudes($solicitudes) { $this->solicitudes = $solicitudes; }
function getSolicitudes() { return $this->solicitudes; }    

 function addPromocion($promocion=null){
     if($promocion!=null){
          if($this->promociones==null){
              $this->promociones=array();   
          }
                
        array_push($this->promociones,$promocion);
          return true;
        }else{
             return false;   
        }
 } 
 function addProducto($producto=null){
     if($producto!=null){
          if($this->productos==null){
              $this->productos=array();   
          }
                
        array_push($this->productos,$producto);
          return true;
        }else{
             return false;   
        }
 } 
 function addSolicitud($solicitud=null){
     if($solicitud!=null){
          if($this->solicitudes==null){
              $this->solicitudes=array();   
          }
                
        array_push($this->solicitudes,$solicitud);
          return true;
        }else{
             return false;   
        }
 }     
function addClasificacion($clasificacion){
  //  echo '<br>llamado agregar clasificacion<br>';
     if($clasificacion!=null){
          if($this->clasificaciones==null){
              $this->clasificaciones=array();   
          }
                
        array_push($this->clasificaciones,$clasificacion);
          return true;
        }else{
             return false;   
        }
 } 

 function addImagen($imagen=null){
     if($imagen!=null){
          if($this->imagenes==null){
              $this->imagenes=array();   
          }
                
        array_push($this->imagenes,$imagen);
          return true;
        }else{
             return false;   
        }
 }     
 function addPreferencia($preferencia=null){
     if($preferencia!=null){
          if($this->preferencias==null){
              $this->preferencias=array();   
          }
                
        array_push($this->preferencias,$preferencia);
          return true;
        }else{
             return false;   
        }
 }     
 function addGeolocalizacion($geolocalizacion=null){
   //  echo '<br>llamado agregar geo<br>';
     if($geolocalizacion!=null){
          if($this->geolocalizaciones==null){
              $this->geolocalizaciones=array();   
          }
                
        array_push($this->geolocalizaciones,$geolocalizacion);
          return true;
        }else{
             return false;   
        }
 }
    
 function addSubscripcion($subscripcion){
     if($subscripcion!=null){
          if($this->subcripciones==null){
             $this->subcripciones=array();   
          }
                
        array_push($this->subcripciones,$subscripcion);
          return true;
        }else{
             return false;   
        }
 }    
    
function generarComercio($json){
    if(isset($json)&&!empty($json)){
         
             $data = json_decode($json,true);
         
            $comer=$data['comercio'];
            
            $this->setIdComercio($comer[0]['idcomercio']);
            $this->setNoComercio($comer[0]['nocomercio']);
            $this->setDiComercio($comer[0]['dicomercio']);
            $this->setPrComercio($comer[0]['prcomercio']);
            $this->setT1Comercio($comer[0]['t1comercio']);
            $this->setT2Comercio($comer[0]['t2comercio']);
            $this->setEmComercio($comer[0]['emcomercio']);
           
            $imgaux=$comer[0]['imagenes'];
            
            foreach($imgaux as $imagen){
              //  echo var_dump($imagen);
                   $imagen1=new EN_Galeria();
                   $imagen1->setIdImagen($imagen['idimagen']);
                   $imagen1->setDtImagen($imagen['dtimagen']);
                   $this->addImagen($imagen1);
            }
            $geolocalizaci=$comer[0]['geolocalizaciones'];
            
            foreach($geolocalizaci as $geolocaliza){
                   $geo=new EN_Geolocalizacion();
                   $geo->setIdlocalizacion($geolocaliza['idlocalizacion']);
                   $geo->setLatitud($geolocaliza['lalocalizacion']);
                   $geo->setLongitud($geolocaliza['lnlocalizacion']);
                   $this->addGeolocalizacion($geo);
            }                 

            $prefaux=$comer[0]['preferencias'];
            
            foreach($prefaux as $pref){
                   $pref1=new EN_Preferencias();
                   $pref1->setIdPreferencia($pref['idpreferencia']);
                   $pref1->setIsContactoSinSubscripcion($pref['contactosinsu']);
                   $pref1->setIsVerProductosSinSubscrib($pref['productosinsu']);
                   $pref1->setHasServicioExpress($pref['servicioexpre']);
                   
                   $this->addPreferencia($pref1);
            }  
                 
            $clasifaux=$comer[0]['clasificaciones'];
            
            foreach($clasifaux as $element){
              
                       $clasificacion1=new EN_CategoriaComercio();
                       $clasificacion1->setIdCategoria($element['idclasificacion']);
                       $clasificacion1->setNoCategoria($element['noclasificacion']);
                  

              $this->addClasificacion($clasificacion1);;    
                    
                
                
            }
            //subscripciones
            $suscrifaux=$comer[0]['subscripciones'];
            
            foreach($suscrifaux as $subscripcion){
                   $subscripcion1=new EN_Subscripcion();
                   $subscripcion1->setIdSubscripcion($subscripcion['idsubscripcion']);
                   $subscripcion1->setIsEnviarPromocion($subscripcion['enviarpromocion']);
                    $user=new EN_Usuario();
               // echo var_dump($subscripcion['usuario']['idusuario']);
                    $user->setIdUsuario($subscripcion['usuario']['idusuario']);
                    $user->setNoUsuario($subscripcion['usuario']['nousuario']);
                    $user->setA1Usuario($subscripcion['usuario']['a1usuario']);
                    $user->setA2Usuario($subscripcion['usuario']['a2usuario']);
                    $user->setFeNacimie($subscripcion['usuario']['fenacimiento']);
                    $user->setSeUsuario($subscripcion['usuario']['sexo']);
                    $subscripcion1->setUsuario($user);
                   // echo var_dump($subscripcion1);
                    $this->addSubscripcion($subscripcion1);
            }
          //productos
            $prodaux=$comer[0]['productos'];
            
            foreach($prodaux as $producto){
              //  echo var_dump($imagen);
                   $producto1=new EN_Producto();
                   $producto1->setIdProducto($producto['idproducto']);
                   $producto1->setNoProducto($producto['noproducto']);
                   $producto1->setDeProducto($producto['deproducto']);
                   $producto1->setPrProducto($producto['precio']);
                   
                   $imgaux=$producto['imagenes'];

                    foreach($imgaux as $imagen){
                      //  echo var_dump($imagen);
                           $imagen1=new EN_Galeria();
                           $imagen1->setIdImagen($imagen['idimagen']);
                           $imagen1->setDtImagen($imagen['dtimagen']);
                           $producto1->addImagen($imagen1);
                    }      
             //   echo count($producto1->getImagenes()).'es la cantidad de imagenes de este producto';
                   $this->addProducto($producto1);
            }
        
            //solicitudes
            $solicitaux=$comer[0]['solicitudes'];
            
            foreach($solicitaux as $solicitud){
              //  echo var_dump($imagen);
                   $solicitud1=new EN_SolicitudCliente();
                   $solicitud1->setIdSolicitud($solicitud['idsolicitud']);
                   $solicitud1->setFeSolicitud($solicitud['fechasolicitud']);
                   $solicitud1->setEsSolicitud($solicitud['estadosolicitud']);
                   $solicitud1->setEHastaSoli($solicitud['esperohasta']);
                   $solicitud1->setCoFInalSoli($solicitud['costofinal']);
                
                    $user=new EN_Usuario();
                    $user->setIdUsuario($solicitud['usuario']['idusuario']);
                    $user->setNoUsuario($solicitud['usuario']['nousuario']);
                    $user->setA1Usuario($solicitud['usuario']['a1usuario']);
                    $user->setA2Usuario($solicitud['usuario']['a2usuario']);
                    $user->setFeNacimie($solicitud['usuario']['fenacimiento']);
                    $user->setSeUsuario($solicitud['usuario']['sexo']);
                    $solicitud1->setUsuario($user);
                    
                    $detalles=$solicitud["detallessolicitud"];
                    
                    foreach($detalles as $producto){
                      //  echo var_dump($imagen);
                           $producto1=new EN_Producto();
                           $producto1->setIdProducto($producto['idproducto']);
                           $producto1->setNoProducto($producto['noproducto']);
                           $producto1->setPrProducto($producto['deproducto']);
                           $producto1->setDeProducto($producto['precio']);
                           $solicitud1->addDetalle($producto1);
                    }
                  
                
                   $this->addSolicitud($solicitud1);
            }

            $promaux=$comer[0]['promociones'];
            
            foreach($promaux as $promocion){
                   $promocion1=new EN_Promocion();
                   $promocion1->setIdPromocion($promocion['idpromocion']);
                   $promocion1->setFechaVencimiento($promocion['fechavencimiento']);
                   $promocion1->setDescripcion($promocion['descripcion']);
                   $promocion1->setTitulo($promocion['titulo']);
                  $promocion1->setFechaPublicacion($promocion['fechapublicacion']);
                 $this->addPromocion($promocion1);
            }          
        
    }
}
}


//$comer=new EN_Comercio();
//$comer->generarComercio($var2);
/*echo '<br>Iniciando ..........................</br>';
//echo $comer->getPrComercio();
foreach($comer->getClasificaciones() as $aux){
  echo '<br>'.$aux->getNoCategoria().'<<<<<br>';   
}

foreach($comer->getImagenes() as $aux){
  echo '<br>imagen '.$aux->getDtImagen().'<<<<<br>';   
}
echo '<br>probando subscriptores<br>';

foreach($comer->getSubcripciones() as $aux){
    echo '<br>id:  '.$aux->getIdSubscripcion();
    echo '<br>enviarpromo:  '.$aux->getIsEnviarPromocion();
    $usuario=$aux->getUsuario();
    echo '<br>user:<br>';
    echo '<br>>:  '.$usuario->getIdUsuario();
    echo '<br>>:  '.$usuario->getNoUsuario();
    echo '<br>>:  '.$usuario->getA1Usuario();
    echo '<br>>:  '.$usuario->getA2Usuario();
    echo '<br>>:  '.$usuario->getFeNacimie();
      
}

echo '<br>probando solicitudes<br>';

foreach($comer->getSolicitudes() as $aux){
    echo '<br>id:  '.$aux->getIdSolicitud();
    echo '<br>fesolicitud:  '.$aux->getFeSolicitud();
    $usuario=$aux->getUsuario();
    echo '<br>user:<br>';
    echo '<br>>:  '.$usuario->getIdUsuario();
    echo '<br>>:  '.$usuario->getNoUsuario();
    echo '<br>>:  '.$usuario->getA1Usuario();
    echo '<br>>:  '.$usuario->getA2Usuario();
    echo '<br>>:  '.$usuario->getFeNacimie();
      
}

echo '<br>Promociones ..........................</br>';
//echo $comer->getPrComercio();
foreach($comer->getPromociones() as $aux){
  echo '<br>'.$aux->getDescripcion().'<<<<<br>';   
}*/


//echo utf8_encode(json_encode($comer));
                 
?>

File: Android/RapiMoncha/src/una/android/rapimoncha/entidades/Comercio.java
package una.android.rapimoncha.entidades;

import android.util.Log;
import org.json.JSONArray;
import org.json.JSONObject;

import una.android.rapimoncha.interfaces.IJsonable;

import java.util.ArrayList;
import java.util.Iterator;

/**
 * Created by Lester on 31/03/2015.
 */
public class Comercio implements IJsonable {
	private int idComercio;
	private String noComerncio;
	private String diComercio;
	private String prComercio;
	private String t1Comercio;
	private String t2Comercio;
	private String emComercio;
	private ArrayList<CategoriaComercio> caComercio;
	private ArrayList<Galeria> imagenes;
	private ArrayList<Subcripcion> subscriptores;
	private ArrayList<GeoLocalizacion> geolocalizaciones;
	private ArrayList<Promocion> promociones;
	private ArrayList<Producto> productos;
	private ArrayList<SolicitudCliente> solicitudes;
	private ArrayList<Preferencia> preferencias;
	private Usuario usuario;

	public Comercio() {
	}

	public int getIdComercio() {
		return idComercio;
	}

	public void setIdComercio(int idComercio) {
		this.idComercio = idComercio;
	}

	public String getNoComerncio() {
		return noComerncio;
	}

	public void limpiarImagenes() {
		if (this.imagenes != null) {

			this.imagenes.clear();
		}

	}

	public void setNoComerncio(String noComerncio) {
		this.noComerncio = noComerncio;
	}

	public String getDiComercio() {
		return diComercio;
	}

	public void setDiComercio(String diComercio) {
		this.diComercio = diComercio;
	}

	public String getPrComercio() {
		return prComercio;
	}

	public void setPrComercio(String prComercio) {
		this.prComercio = prComercio;
	}

	public String getT1Comercio() {
		return t1Comercio;
	}

	public void setT1Comercio(String t1Comercio) {
		this.t1Comercio = t1Comercio;
	}

	public String getT2Comercio() {
		return t2Comercio;
	}

	public void setT2Comercio(String t2Comercio) {
		this.t2Comercio = t2Comercio;
	}

	public String getEmComercio() {
		return emComercio;
	}

	public void setEmmComercio(String emmComercio) {
		this.emComercio = emmComercio;
	}

	public ArrayList<CategoriaComercio> getCaComercio() {
		return caComercio;
	}

	public void setCaComercio(ArrayList<CategoriaComercio> caComercio) {
		this.caComercio = caComercio;
	}

	public ArrayList<Galeria> getImagenes() {
		return imagenes;
	}

	public void setImagenes(ArrayList<Galeria> imagenes) {
		this.imagenes = imagenes;
	}

	public ArrayList<Subcripcion> getSubscriptores() {
		return subscriptores;
	}

	public void setSubscriptores(ArrayList<Subcripcion> subscriptores) {
		this.subscriptores = subscriptores;
	}

	public ArrayList<GeoLocalizacion> getUbicacionmapa() {
		return geolocalizaciones;
	}

	public void setUbicacionmapa(ArrayList<GeoLocalizacion> ubicacionmapa) {
		this.geolocalizaciones = ubicacionmapa;
	}

	public ArrayList<Promocion> getPromocion() {
		return promociones;
	}

	public void setPromocion(ArrayList<Promocion> promocion) {
		this.promociones = promocion;
	}

	public ArrayList<Producto> getProductos() {
		return productos;
	}

	public void setProductos(ArrayList<Producto> productos) {
		this.productos = productos;
	}

	public ArrayList<SolicitudCliente> getSolicitudes() {
		return solicitudes;
	}

	public void setSolicitudes(ArrayList<SolicitudCliente> solicitudes) {
		this.solicitudes = solicitudes;
	}

	public void addPromocion(Promocion promocion) {
		if (promociones == null) {
			promociones = new ArrayList<Promocion>();
		}

		promociones.add(promocion);
	}

	public void addProducto(Producto producto) {
		if (productos == null) {
			productos = new ArrayList<Producto>();
		}

		productos.add(producto);
	}

	public void addSubscripcion(Subcripcion subcripcion) {
		if (subscriptores == null) {
			subscriptores = new ArrayList<Subcripcion>();
		}

		subscriptores.add(subcripcion);
	}

	public void addSolicitud(SolicitudCliente solicitud) {
		if (solicitudes == null) {
			solicitudes = new ArrayList<SolicitudCliente>();
		}

		solicitudes.add(solicitud);
	}

	public void addClasificacion(CategoriaComercio clasificacion) {
		if (this.caComercio == null) {
			caComercio = new ArrayList<CategoriaComercio>();
		}

		caComercio.add(clasificacion);
	}

	public void limpiarClasificaciones() {

		if (this.caComercio == null) {
			caComercio = new ArrayList<CategoriaComercio>();
		}

		caComercio.clear();

	}

	public void addImagen(Galeria imagen) {
		if (imagenes == null) {
			imagenes = new ArrayList<Galeria>();
		}

		imagenes.add(imagen);
	}

	public void limpiarPreferencias() {

		if (this.preferencias == null) {
			preferencias = new ArrayList<Preferencia>();
		}

		preferencias.clear();

	}

	public void addPreferencia(Preferencia preferencia) {
		if (preferencias == null) {
			preferencias = new ArrayList<Preferencia>();
		}

		preferencias.add(preferencia);
	}

	public ArrayList<Preferencia> getPreferenciascomercio() {
		return this.preferencias;
	}

	public void addGeolocalizacion(GeoLocalizacion geolocalizacion) {
		if (geolocalizaciones == null) {
			geolocalizaciones = new ArrayList<GeoLocalizacion>();
		}

		geolocalizaciones.add(geolocalizacion);
	}

	@Override
	public String toString() {
		String resp = this.t2Comercio
				+ (this.geolocalizaciones != null ? this.geolocalizaciones
						.size() : "geofalso")
				+ (this.caComercio != null ? this.caComercio.size()
						: "categorias falso") + (noComerncio);

		return resp;
	}

	@Override
	public JSONObject getJson() {
		JSONObject finale = new JSONObject();

		try {
			JSONObject json_comercio = new JSONObject();
			json_comercio.put("idcomercio", this.idComercio);
			json_comercio.put("nocomercio", this.noComerncio);
			json_comercio.put("dicomercio", this.diComercio);
			json_comercio.put("prcomercio", this.prComercio);
			json_comercio.put("t1comercio", this.t1Comercio);
			json_comercio.put("t2comercio", this.t2Comercio);
			json_comercio.put("emcomercio", this.emComercio);

			// usuario	
			if (usuario != null) {
				JSONArray js_usuariocomer = new JSONArray();
				JSONObject obauxa = usuario.getJson();
				js_usuariocomer.put(obauxa);
				json_comercio.put("usuario", js_usuariocomer);
			}
			// insertar imagenes
			Iterator<Galeria> iter_galeria = this.imagenes != null ? this.imagenes
					.iterator() : null;
			JSONArray js_imagenescomer = new JSONArray();
			if (iter_galeria != null) {
				Log.i("Mi nota", "Se agregaron se agrego imagenes");
				while (iter_galeria.hasNext()) {
					Galeria aux = iter_galeria.next();
					JSONObject obaux = aux.getJson();
					js_imagenescomer.put(obaux);
				}
			} else {
				Log.i("Mi nota", "No se agrego imagenes");
			}
			json_comercio.put("imagenes", js_imagenescomer);

			// insertar geolozcalizacion
			Iterator<GeoLocalizacion> iter_geolocalizacion = this.geolocalizaciones != null ? this.geolocalizaciones
					.iterator() : null;
			JSONArray js_geolocalizacioncomer = new JSONArray();
			if (iter_geolocalizacion != null) {

				while (iter_geolocalizacion.hasNext()) {
					GeoLocalizacion aux = iter_geolocalizacion.next();
					JSONObject obaux = aux.getJson();
					js_geolocalizacioncomer.put(obaux);
				}
			}
			json_comercio.put("geolocalizaciones", js_geolocalizacioncomer);

			// insertar preferencias
			Iterator<Preferencia> iter_preferencia = this.preferencias != null ? this.preferencias
					.iterator() : null;
			JSONArray js_preferencias = new JSONArray();
			if (iter_preferencia != null) {

				while (iter_preferencia.hasNext()) {
					Preferencia aux = iter_preferencia.next();
					JSONObject obaux = aux.getJson();
					js_preferencias.put(obaux);
				}
			}
			json_comercio.put("preferencias", js_preferencias);

			// insertar clasificaciones
			Iterator<CategoriaComercio> iter_clasificaciones = this.caComercio != null ? this.caComercio
					.iterator() : null;
			JSONArray js_clasificaciones = new JSONArray();
			if (iter_clasificaciones != null) {

				while (iter_clasificaciones.hasNext()) {
					CategoriaComercio aux = iter_clasificaciones.next();
					JSONObject obaux = aux.getJson();
					js_clasificaciones.put(obaux);
				}
			}
			json_comercio.put("clasificaciones", js_clasificaciones);

			// insertar subscripciones
			Iterator<Subcripcion> iter_subscripciones = this.subscriptores != null ? this.subscriptores
					.iterator() : null;
			JSONArray js_subscripciones = new JSONArray();
			if (iter_subscripciones != null) {
				while (iter_subscripciones.hasNext()) {
					Subcripcion aux = iter_subscripciones.next();
					JSONObject obaux = aux.getJson();
					js_subscripciones.put(obaux);
				}
			}
			json_comercio.put("subscripciones", js_subscripciones);

			// insertar productos
			Iterator<Producto> iter_productos = this.productos != null ? this.productos
					.iterator() : null;
			JSONArray js_productos = new JSONArray();
			if (iter_productos != null) {
				while (iter_productos.hasNext()) {
					Producto aux = iter_productos.next();
					JSONObject obaux = aux.getJson();
					js_productos.put(obaux);
				}
			}
			json_comercio.put("productos", js_productos);

			// insertar solicitudes cliente
			Iterator<SolicitudCliente> iter_solicitudes = this.solicitudes != null ? this.solicitudes
					.iterator() : null;
			JSONArray js_solicitudes = new JSONArray();
			if (iter_solicitudes != null) {

				while (iter_solicitudes.hasNext()) {
					SolicitudCliente aux = iter_solicitudes.next();
					JSONObject obaux = aux.getJson();
					js_solicitudes.put(obaux);
				}
			}
			json_comercio.put("solicitudes", js_solicitudes);

			// insertar clasificaciones
			Iterator<Promocion> iter_promociones = this.promociones != null ? this.promociones
					.iterator() : null;
			JSONArray js_promociones = new JSONArray();
			if (iter_promociones != null) {
				while (iter_promociones.hasNext()) {
					Promocion aux = iter_promociones.next();
					JSONObject obaux = aux.getJson();
					js_promociones.put(obaux);
				}
			}
			json_comercio.put("promociones", js_promociones);
			JSONArray json_comercio_ = new JSONArray();
			json_comercio_.put(json_comercio);
			finale.put("comercio", json_comercio_);

			Log.i("resultado", finale.toString());
		} catch (Exception ex) {
			ex.printStackTrace();
		}
		return finale;
	}

	public Usuario getUsuario() {
		return usuario;
	}

	public void setUsuario(Usuario usuario) {
		this.usuario = usuario;
	}
}
